src/WebJobs.Script/StorageProvider/HostAzureTableStorageProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using Azure.Core.Pipeline;
+using Azure.Data.Tables;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    internal class HostAzureTableStorageProvider : IAzureTableStorageProvider
+    {
+        private readonly ILogger<HostAzureTableStorageProvider> _logger;
+        private readonly IDelegatingHandlerProvider _delegatingHandlerProvider;
+        private readonly TableServiceClientProvider _tableServiceClientProvider;
+
+        public HostAzureTableStorageProvider(IConfiguration configuration, ILogger<HostAzureTableStorageProvider> logger, AzureComponentFactory componentFactory, AzureEventSourceLogForwarder logForwarder, IDelegatingHandlerProvider delegatingHandlerProvider)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _delegatingHandlerProvider = delegatingHandlerProvider;
+            _tableServiceClientProvider = new TableServiceClientProvider(componentFactory, logForwarder);
+            Configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        }
+
+        public virtual IConfiguration Configuration { get; private set; }
+
+        public virtual bool TryCreateTableServiceClientFromConnection(string connection, out TableServiceClient client)
+        {
+            var connectionToUse = connection ?? ConnectionStringNames.Storage;
```

Comment:
Since we have the other overload which uses the default ( which is `ConnectionStringNames.Storage`). Should we just throw if connection parameter is null here?